installers/AppVeyor.hs

```diff
@@ -0,0 +1,160 @@
+artifactForJobId :: Text -> IO [(Text, L8.ByteString)]
+artifactForJobId jobId = do
+  let req = jobArtifactsURI jobId
+  r <- asValue =<< get req
+  forM (r ^.. responseBody . artifactFilenames) $ \filename -> do
+    let req' = req ++ ('/':T.unpack filename)
+    printf ("Downloading "%w%" ... ") req'
+    r' <- get req'
+    printf "done\n"
+    pure (filename, r' ^. responseBody)
+
+-- | Looks into AppVeyor API build info json to get job IDs
+buildJobs :: Monoid r => Getting r Value Job
+buildJobs = key "build" . key "jobs" . _Array . traverse . _JSON
+
+-- | Looks into AppVeyor API job artifacts json to get the filenames
+artifactFilenames :: Monoid r => Getting r Value Text
+artifactFilenames = _Array . traverse . key "fileName" . _String
+
+-- | The AppVeyor API URL of an build, given its web URL.
+buildAPI :: GH.URL -> Maybe URI
+buildAPI = parseAbsoluteURI . T.unpack . T.replace "/project/" "/api/projects/" . getUrl
+
+-- | The AppVeyor API URL to list artifacts belonging to a job.
+jobArtifactsURI :: Text -> String
+jobArtifactsURI jobId = "https://ci.appveyor.com/api/buildJobs/" <> T.unpack jobId <> "/artifacts"
+
+-- | Use GitHub API to find AppVeyor build status
+appVeyorURL :: CardanoSource -> IO GH.URL
+appVeyorURL src = fmap collect <$> statusFor' src >>= \case
+  Right (Just u) -> pure u
+  Right Nothing  -> throwM $ StatusMissingError src
+  Left err       -> throwM $ GitHubStatusError src err
+  where
+    collect = head . mapMaybe statusTargetUrl . filter isAppVeyor .
+              toList . combinedStatusStatuses
+    isAppVeyor st = statusContext st == Just "continuous-integration/appveyor/branch"
+
+statusFor' :: CardanoSource -> IO (Either GH.Error GH.CombinedStatus)
+statusFor' CardanoSource{..} = statusFor noAuth srcOwner srcRepo srcRev
+  where noAuth = BasicAuth "" ""
+
+data JobStatus = JobSuccess | JobFailed | JobRunning
+               | JobUnknown Text deriving (Show, Eq, Generic)
+
+data Job = Job { jobId :: Text
+               , jobStatus :: JobStatus
+               } deriving (Show, Eq, Generic)
+
+instance FromJSON JobStatus where
+  parseJSON = withText "status" $ \case
+    "success" -> pure JobSuccess
+    "failed" -> pure JobFailed
+    "running" -> pure JobRunning
+    other -> pure $ JobUnknown other
+
+instance FromJSON Job where
+  parseJSON = withObject "Job" $ \o -> Job <$> o .: "jobId" <*> o .: "status"
+
+instance ToJSON JobStatus
+instance ToJSON Job -- Only required for _JSON prism
+
+----------------------------------------------------------------------------
+
+-- | Load commit hash and GitHub repo name from cardano-sl-src.json.
+getCardanoRev :: FilePath -> IO CardanoSource
+getCardanoRev src = eitherDecode <$> L8.readFile src >>= \case
+  Right rev -> pure rev
+  Left err -> throwM (SourceJSONDecodeError err)
+
+data CardanoSource = CardanoSource
+  { srcOwner :: Name Owner
+  , srcRepo  :: Name Repo
+  , srcRev   :: Name Commit
+  } deriving (Show, Eq)
```

Comment:
Does this type need to be specific to Cardano -- is it actually a Github repo source, in disguise?

Reply:
What do you propose?

Reply:
We already have something like this: https://github.com/input-output-hk/iohk-ops/blob/develop/iohk/Nix.hs#L29, which is more generic in purpose.  We could use something similar here, without being tied to Cardano specifics.

Reply:
I suggest we factor out this later on into a library to handle pinning in such style.